New materials dev2

RogerJBarlow opened the pull request:
for the release for the deliverable

Sam-Tygier commented:
Clean ups
`cu50_test.cpp`
Comments and extra couts. Lines 120 and 264

`ScatteringModel.h/cpp`
If we bring back `ScatteringModelMerlin` and `ScatteringModelSixTrack`, they only need `Configure()` to be virtual. I think as they are currently they will do something odd by calling `Processes.push_back(S)`.

Shouldn't be much work, but would save an API change.
I am happy to do this. After merging.


`ScatteringModel::PathLength`
Do we need to keep old version in comments, if we need it back we can find it in git.

news and deletes. probably some good use cases for unique pointers. I can investigate later.

`ScatteringModelsMerlin.cpp`
`AddProcess(new ElasticpN(7000.0));`
looks bad, but would be cleaned up by above mentioned fixes

`ScatteringProcess.cpp`
A few commented out functions.
Can we deleted them

Roger, if you make changes, then commit and push then they should automatically be added to this pull request.

RogerJBarlow commented:
Clean ups
cu50_test.cpp
Comments and extra couts. Lines 120 and 264

<<< Yes, couts removed

ScatteringModel.h/cpp
If we bring back ScatteringModelMerlin and ScatteringModelSixTrack, they only need Configure() to be virtual. I think as they are currently they will do something odd by calling Processes.push_back(S).

<< These two should now be redundant

Shouldn't be much work, but would save an API change.
I am happy to do this. After merging.

ScatteringModel::PathLength
Do we need to keep old version in comments, if we need it back we can find it in git.

<<. Probably safer, yes.

news and deletes. probably some good use cases for unique pointers. I can investigate later.

<< Yes.   Add to list

ScatteringModelsMerlin.cpp
AddProcess(new ElasticpN(7000.0));
looks bad, but would be cleaned up by above mentioned fixes

<<<<<.  or removal

ScatteringProcess.cpp
A few commented out functions.
Can we deleted them

<<<<<. I commented them out rather than deleting them 'just in case'
<<<<<  I know this is cowardly.    I'd rather delete them properly next time

Roger, if you make changes, then commit and push then they should automatically be added to this pull request.

<<<<<<<. I've changed cu50_test.cpp and committed and pushed

Roger Barlow
 Emeritus Professor

________________________________________
From: Sam-Tygier <notifications@github.com>
Sent: 01 March 2020 17:48
To: Merlin-Collaboration/Merlin
Cc: Roger Barlow; Author
Subject: Re: [Merlin-Collaboration/Merlin] New materials dev2 (#47)

Clean ups
cu50_test.cpp
Comments and extra couts. Lines 120 and 264

ScatteringModel.h/cpp
If we bring back ScatteringModelMerlin and ScatteringModelSixTrack, they only need Configure() to be virtual. I think as they are currently they will do something odd by calling Processes.push_back(S).

Shouldn't be much work, but would save an API change.
I am happy to do this. After merging.

ScatteringModel::PathLength
Do we need to keep old version in comments, if we need it back we can find it in git.

news and deletes. probably some good use cases for unique pointers. I can investigate later.

ScatteringModelsMerlin.cpp
AddProcess(new ElasticpN(7000.0));
looks bad, but would be cleaned up by above mentioned fixes

ScatteringProcess.cpp
A few commented out functions.
Can we deleted them

Roger, if you make changes, then commit and push then they should automatically be added to this pull request.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://github.com/Merlin-Collaboration/Merlin/pull/47?email_source=notifications&email_token=AHNQ6N2EY24G6RWAAPODNR3RFKNVPA5CNFSM4K7GK23KYY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOENNFYEQ#issuecomment-593124370>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AHNQ6N7UQ3GWDOMFUC46UH3RFKNVPANCNFSM4K7GK23A>.
University of Huddersfield inspiring global professionals.
[http://marketing.hud.ac.uk/_HOSTED/EmailSig2014/EmailSigFooterMarch2019.jpg]

This transmission is confidential and may be legally privileged. If you receive it in error, please notify us immediately by e-mail and remove it from your system. If the content of this e-mail does not relate to the business of the University of Huddersfield, then we do not endorse it and will accept no liability.
